### ISSUE TITLE
Can the author provide some examples for the input_file

### ISSUE THREAD

**Issue by LiuAlex1109** [AkariAsai/OpenScholar#5]

**Comment by Zailushang211** [comment]
Same request. Very difficult to run the codes.
1. can not get api key from  Semantic: "Please note that our API key issuance is currently on pause due to high demand".
2. After tried for 2 days, stuck at the process_input_data:
 '''
def process_input_data(data, use_contexts=True):
    processed_data = []
    for item in data:
        if "answer" not in item:
            item["answer"] = ""
        if "input" not in item:
            if "question" in item:
                item["input"] = item["question"]
            if "query" in item:
                item["input"] = item["query"]

        new_ctxs = []
        if use_contexts is True:
            # normalize ctx format for different retrieval APIs
            for ctx in item["ctxs"]:
                if type(ctx) is list:
                    for c in ctx:
                        if type(c) is dict:
                            new_ctxs.append(c)
                if type(ctx) is dict:
                    new_ctxs.append(ctx)
            item["ctxs"] = new_ctxs
'''
anyone know how to set the ctxs in the input file? seems this must be set True, 
parser.add_argument("--use_contexts", action="store_true", help="set True whenever you use RAG pipelines.")

**Comment by abdul-456** [comment]
@Zailushang211 were you able to run the codes?

**Comment by tian969** [comment]
I solved this problem by: 
1. set the api_key = '' You need not use when use a standard RAG;
2. make a json like this: {"query":"", "ctxs":[]}  and it will skip the check of ctxs.
@abdul-456 @Zailushang211

**Comment by tian969** [comment]
I think we should run the retriever first to get the input_file. and you can choose not to use the semantic api and set the api_key = ''. @Zailushang211 @abdul-456

**Comment by Zailushang211** [comment]
> @Zailushang211 were you able to run the codes?

not yet

**Comment by Zailushang211** [comment]
> I think we should run the retriever first to get the input_file. and you can choose not to use the semantic api and set the api_key = ''. @Zailushang211 @abdul-456

As far as i can see, the retriever is used to build the database index for searching the top N related papers/passages, nothing about the input_file, the input file should be a json file with contents like {"query": "deep learning methods for image generation"}, I tested with this format, but logs showed that the input needs ctx.

**Comment by tian969** [comment]
I tried the retrieval section's code, and get:
{"query": "What did Einstein invent?", "raw_query": "What did Einstein invent?", "ctxs": [{"id": "1024249", "source": "pes2o_v2", "retrieval text": "know why he won his Nobel Prize? (It was not for relativity, but for the photoelectric effect in 1921.) We have but a fuzzy view of his work;  ....]}
**It seems that ctxs, the contexts, are the retrieved passages in the specific format**. So it will work when you run the retrieval first. 
@Zailushang211 @LiuAlex1109 @abdul-456 @AkariAsai

**Comment by Presence76** [comment]
They said in the instructions 
By default, OpenScholar takes retrieval results from off-line retrieval results after running the retrieval scripts in [retrieval/](https://github.com/AkariAsai/OpenScholar/blob/main/retireval), followed by additional retrieval from Semantic Scholar Paper API and web search API results. See the script [src/use_search_apis.py](https://github.com/AkariAsai/OpenScholar/blob/main/src/use_search_apis.py) to retrieve related passages offline using external APIs
We released our retrieval results at [google drive](https://drive.google.com/drive/folders/1lOloYPOveKesD-37lD4Dlju96tc0XIm9?usp=sharing).
The retrieval results are the example data basically.

**Comment by Presence76** [comment]
> I tried the retrieval section's code, and get: {"query": "What did Einstein invent?", "raw_query": "What did Einstein invent?", "ctxs": [{"id": "1024249", "source": "pes2o_v2", "retrieval text": "know why he won his Nobel Prize? (It was not for relativity, but for the photoelectric effect in 1921.) We have but a fuzzy view of his work; ....]} **It seems that ctxs, the contexts, are the retrieved passages in the specific format**. So it will work when you run the retrieval first. @Zailushang211 @LiuAlex1109 @abdul-456 @AkariAsai
I wonder if you successfully made a workflow to use just a single input query for inference, I am working on it but can not make it work

**Comment by tian969** [comment]
@Presence76  Yes. I have run the traditional RAG and RAG with feedback successfully with this input file. You should read the README in retriever and run the code building index and offline search, and you'll get a  input_file with corresponding ctxs. then you can run the run.py related code!

**Comment by tian969** [comment]
I use a single H100 80G GPU, and I'm not pay attention to the memory useage. In theory you can use tensors parallel in the VLLM section, or you can use a bigger GPU to run with feedback.

**Comment by Zailushang211** [comment]
> I tried the retrieval section's code, and get: {"query": "What did Einstein invent?", "raw_query": "What did Einstein invent?", "ctxs": [{"id": "1024249", "source": "pes2o_v2", "retrieval text": "know why he won his Nobel Prize? (It was not for relativity, but for the photoelectric effect in 1921.) We have but a fuzzy view of his work; ....]} **It seems that ctxs, the contexts, are the retrieved passages in the specific format**. So it will work when you run the retrieval first. @Zailushang211 @LiuAlex1109 @abdul-456 @AkariAsai

Thanks a lot, will try again with your instruction.

**Comment by abdul-456** [comment]
Hi guys, did anyone tried to full finetune any other llm using the openscholar training data through torchtune? If yes then how did you run it? I followed the instructions provided in the repo, but inferencing the finetuned model gives me the following error:

/build/pytorch-WT7PzF/pytorch-2.5.1+ds/aten/src/ATen/native/cuda/TensorCompare.cu:110: _assert_async_cuda_kernel: block: [0,0,0], thread: [0,0,0] Assertion `probability tensor contains either `inf`, `nan` or element < 0` failed.

does anyone know a work around it?

thanks

**Comment by Presence76** [comment]
> I use a single H100 80G GPU, and I'm not pay attention to the memory useage. In theory you can use tensors parallel in the VLLM section, or you can use a bigger GPU to run with feedback.

How did you set up the environment for the retriever. The environment.yml file in the retriever just does not work.

**Comment by Presence76** [comment]
@tian969 Hi I want to ask what is the exact pipeline you used to run the retriever, I can not use it for offline search and got some error about shape of the index and raw text mismatch after trying for a long time. I basically download their embeddings and raw passages on their huggingface dataset. It seems is causing the problems. I also try to build the embedding myself but the building process does not stop correctly. I am following everything they put up on the git hub page and failed

**Comment by LouisDo2108** [comment]
Hi @tian969 and @Zailushang211, did you encounter this problem during search?
```
assert len(passages) == index.index.ntotal, f"number of documents {len(passages)} and number of embeddings {index.index.ntotal} mismatch"
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: number of documents 14905277 and number of embeddings 14646940 mismatch
```

I tried to index with the provided OpenSciLM--OpenScholar-DataStore-V3 embeddings, and this is the error from searching only on the first shard passages_00.pkl and raw_passages-0-of-16.jsonl
